### PR TITLE
DM-37051: Drop now-unnecessary types from docs

### PR DESCRIPTION
**Pull request by rra**
The current documenteer setup correctly resolves types and doesn't require reiterating them in docstrings in most cases.  Drop them where possible.  Add better docstrings to datetime_from_db and datetime_to_db.

For reasons that I don't understand, structlog.types.EventDict doesn't resolve unquoted but does work in backticks, so add backticks to keep Sphinx happy.

### CODE REVIEW

**Comment by rra** [comment]
Sorry about the review request noise. For some reason GitHub was showing no requested reviewers, and then removed you when I tried to add you.

File: src/safir/database.py [hunk 1]
```diff
@@ -91,7 +91,19 @@ def datetime_from_db(time: None) -> None:
 
 
 def datetime_from_db(time: Optional[datetime]) -> Optional[datetime]:
-    """Add the UTC time zone to a naive datetime from the database."""
+    """Add the UTC time zone to a naive datetime from the database.
+
+    Parameters
+    ----------
+    time
+        The naive datetime from the database, or `None`
```
Comment (jonathansick) [inline src/safir/database.py:99]:
```suggestion
        The naive datetime from the database, or `None`.
```

File: src/safir/database.py [hunk 2]
```diff
@@ -91,7 +91,19 @@ def datetime_from_db(time: None) -> None:
...
+
+    Returns
+    -------
+    datetime.datetime or None
```
Comment (jonathansick) [inline src/safir/database.py:103]: I noticed that the `datetime.datetime` gets linked, but the `None` doesn't. I think it's fine tbh, and explicitly marking up the `None` may not be worth it. I guess the rule is that the first "word" gets auto marked up, and the remaining words are treated as text.

File: src/safir/database.py [hunk 3]
```diff
@@ -110,7 +122,21 @@ def datetime_to_db(time: None) -> None:
 
 
 def datetime_to_db(time: Optional[datetime]) -> Optional[datetime]:
-    """Strip time zone for storing a datetime in the database."""
+    """Strip time zone for storing a datetime in the database.
```
Comment (jonathansick) [inline src/safir/database.py:125]: Oh yeah, the treatment of overloads is great.

**Review by jonathansick — APPROVED** [review]
Excellent, thanks!

File: src/safir/database.py (hunk 2 above)
Comment (rra) [reply src/safir/database.py:103]: It's a bit more complex in my previous experiments: if you mark up `None`, then `datetime.datetime` doesn't get linked (unless you also mark it up). It's kind of weird.

**Pull request merged**